### ISSUE TITLE
studio terminal bug in 0.12.0

### ISSUE THREAD

**Issue by juliandunn** [habitat-sh/habitat#1435]
It looks like we have a studio terminal handling bug inside of 0.12.0 studios. On 0.11.0, if you enter a studio and try to backspace after typing things, it worked. Now, trying to backspace instead adds an additional space to the input.

Additionally, if you try to run `reset` to reset your terminal, you get the following error:

```
reset: can't initialize terminal type xterm (error -1)
Terminal type?
```

(with an interactive prompt)

I'm suspecting the bug is related to #1318. The studio is correctly built with `core/ncurses` inside it, but there is no such directory `/hab/pkgs/core/ncurses/6.0/20161031042533/share/terminfo`.

**Comment by juliandunn** [comment]
A workaround is to install the older version of `core/ncurses` and set the `TERMINFO` database to it.

`hab pkg install core/ncurses/6.0/20160612075116`
`export TERMINFO=/hab/pkgs/core/ncurses/6.0/20160612075116/share/terminfo`

**Comment by fnichol** [comment]
Okay, it turns out that having a `TERMINFO` environment variable set affects the build of `core/ncurses` without changing the plan at all--cool no?

Rather than re-evaluating #1318 I think our way out is to explicitly `unset TERMINFO` in the build program to guarantee that this userland convenience never leaks into a plan build. I'm not entirely sure what other software or autoconf tooling might do with this environment variable but would rather provide as minimalistic a starting point as possible with respect to environment variables.

I'll have a PR shortly to the Studio codebase to unset this variable which is the code fix, but we'll want to rebuild ncurses so that it contains the terminfo database entries as before. Good thing is: I'm pretty fresh on this activity 😉

**Comment by juliandunn** [comment]
fixed in #1440
